runner_service/controllers/playbooks.py

```diff
@@ -145,59 +145,62 @@ def _run_playbook(playbook_name, tags=None):
                           "Invalid content-type({}). Use application/" \
                           "json".format(request.content_type)
         return r
-
-    vars = request.get_json()
-    filter = request.args.to_dict()
-    if not all([_k in valid_filter for _k in filter.keys()]):
-        r.status, r.msg = "INVALID", "Bad request, supported " \
-                          "filters are: {}".format(','.join(valid_filter))
-        return r
-
-    if 'limit' in filter:
-
-        target_hosts = filter['limit'].split(',')
-        inv_hosts = AnsibleInventory().hosts
-        logger.debug("Checking host limit against the inventory")
-        if all([_h in inv_hosts for _h in target_hosts]):
-            logger.debug("hosts in the limit list match the inventory")
-
-        else:
-            logger.error("limit hosts don't match with the inventory")
-            # host(s) provided are not all in the inventory
-            r.status, r.msg = "INVALID", \
-                              "Host(s) provided not in Ansible inventory"
+    try:
+        vars = request.get_json()
+    except Exception as e:
+        logger.error("Failed to decode JSON object: {}".format(e))
```

Comment:
Instead of moving everything to `else` section, wouldn't it be more readable to return immediately?

```
    try:
        vars = request.get_json()
    except Exception as e:
        r.status, r.msg = "INVALID", "Failed to decode JSON object: {}".format(e))
        logger.error(r)
        return r
```

Reply:
done, thanks